mlrun/datastore/datastore_profile.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import ast
+import base64
+import json
+import typing
+from urllib.parse import ParseResult, urlparse, urlunparse
+
+import pydantic
+
+import mlrun
+import mlrun.errors
+
+from ..secrets import get_secret_or_env
+
+
+class privateValue(pydantic.BaseModel):
+    value: str
+
+    def get(self):
+        if self.value == "None":
+            return None
+        return ast.literal_eval(self.value)["value"]
+
+
+class DatastoreProfile(pydantic.BaseModel):
+    type: str
+    name: str
+
+    @pydantic.validator("name")
+    def lower_case(cls, v):
+        return v.lower()
+
+    @staticmethod
+    def generate_secret_key(profile_name: str, project: str):
+        secret_name_separator = "-__-"
+        full_key = (
+            "datastore-profiles"
+            + secret_name_separator
+            + project
+            + secret_name_separator
+            + profile_name
+        )
+        return full_key
+
+
+class DatastoreProfileRedis(DatastoreProfile):
+    type: str = pydantic.Field("redis")
+    endpoint_url: str
+    username: typing.Optional[privateValue]
+    password: typing.Optional[privateValue]
+
+    @pydantic.validator("username", "password", pre=True)
+    def convert_to_private(cls, v):
```

Comment:
IDE complains about this not being a class method. I guess this is in order?
```suggestion
    def convert_to_private(self, v):
```

Reply:
pydantic requires cls